core/src/context/ctx.rs

```diff
@@ -55,10 +72,28 @@ impl<'js> Ctx<'js> {
         handle_exception(self, val)
     }
 
-    /// Evaluate a script in global context
+    /// Evaluate a script in global context.
     pub fn eval<V: FromJs<'js>, S: Into<Vec<u8>>>(self, source: S) -> Result<V> {
+        self.eval_with_options(source, Default::default())
+    }
+
+    /// Evaluate a script with the given options.
+    pub fn eval_with_options<V: FromJs<'js>, S: Into<Vec<u8>>>(
+        self,
+        source: S,
+        options: EvalOptions,
+    ) -> Result<V> {
         let file_name = unsafe { CStr::from_bytes_with_nul_unchecked(b"eval_script\0") };
-        let flag = qjs::JS_EVAL_TYPE_GLOBAL | qjs::JS_EVAL_FLAG_STRICT;
+        let mut flag = 0;
+
+        if options.global {
+            flag |= qjs::JS_EVAL_TYPE_GLOBAL;
+        }
```

Comment:
In that context the opposite of `global` flags is `module` so in my opinion this block should looks like the following:
```rust
flag |= if options.global { qjs::JS_EVAL_TYPE_GLOBAL } else { qjs::JS_EVAL_TYPE_MODULE };
```
Otherwise this block is useless because the `JS_EVAL_TYPE_GLOBAL` flag is zero.
Other variants is out of scope because it is for internal use only.